homeassistant/components/lutron_caseta/light.py

```diff
@@ -48,37 +53,145 @@ async def async_setup_entry(
+        elif light["type"] == "WhiteTune":
+            self._attr_supported_color_modes = {
+                ColorMode.COLOR_TEMP,
+                ColorMode.BRIGHTNESS,
+                ColorMode.WHITE,
+            }
+        else:
+            self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
 
-    _attr_color_mode = ColorMode.BRIGHTNESS
-    _attr_supported_color_modes = {ColorMode.BRIGHTNESS}
     _attr_supported_features = LightEntityFeature.TRANSITION
 
     @property
-    def brightness(self):
+    def brightness(self) -> int:
         """Return the brightness of the light."""
         return to_hass_level(self._device["current_state"])
 
-    async def _set_brightness(self, brightness, **kwargs):
+    async def _set_brightness(self, brightness, color_value, **kwargs):
         args = {}
         if ATTR_TRANSITION in kwargs:
             args["fade_time"] = timedelta(seconds=kwargs[ATTR_TRANSITION])
 
+        if brightness is not None:
+            brightness = to_lutron_level(brightness)
         await self._smartbridge.set_value(
-            self.device_id, to_lutron_level(brightness), **args
+            self.device_id, value=brightness, color_value=color_value, **args
         )
 
+    async def _set_warm_dim(self, brightness, **kwargs):
+        """Set the light to warm dim mode."""
+        args = {}
+        if ATTR_TRANSITION in kwargs:
+            args["fade_time"] = timedelta(seconds=kwargs[ATTR_TRANSITION])
+
+        if brightness is not None:
+            brightness = to_lutron_level(brightness)
+
+        await self._smartbridge.set_warm_dim(self.device_id, brightness, **args)
+
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the light on."""
-        brightness = kwargs.pop(ATTR_BRIGHTNESS, 255)
-
-        await self._set_brightness(brightness, **kwargs)
+        # first check for "white mode" (WarmDim)
+        white_color = kwargs.pop(ATTR_WHITE, None)
+        if white_color is not None:
+            await self._set_warm_dim(white_color)
+            return
+
+        brightness = kwargs.pop(ATTR_BRIGHTNESS, None)
+        color = None
+        hs_color = kwargs.pop(ATTR_HS_COLOR, None)
+        kelvin_color = kwargs.pop(ATTR_COLOR_TEMP_KELVIN, None)
+        if hs_color is not None:
+            color = FullColorValue(hs_color[0], hs_color[1])
+        elif kelvin_color is not None:
+            color = WarmCoolColorValue(kelvin_color)
+
+        # if user is pressing on button nothing is set, so set brightness to 255
+        if color is None and brightness is None:
+            brightness = 255
+
+        await self._set_brightness(brightness, color, **kwargs)
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the light off."""
-        await self._set_brightness(0, **kwargs)
+        await self._set_brightness(0, None, **kwargs)
+
+    @property
+    def color_mode(self) -> ColorMode:
+        """Return the current color mode of the light."""
```

Comment:
https://developers.home-assistant.io/docs/core/entity/light?_highlight=light#color-modes

Please double check to make sure all color modes match the description in the table

Reply:
Ah, I see brightness is suposed to be standalone, and you should only return values in supported_color_modes. 
Would this be ok, is is there something else that needs to be fixed?

```

     if light["type"] == "SpectrumTune":
            self._attr_supported_color_modes = {
                ColorMode.HS,
                ColorMode.COLOR_TEMP,
                ColorMode.WHITE,
            }

................................................................................................................


    @property
    def color_mode(self) -> ColorMode:
        """Return the current color mode of the light."""

        current_color = self._device.get("color")
        if ColorMode.BRIGHTNESS in self._attr_supported_color_modes:
            return ColorMode.BRIGHTNESS
        elif ColorMode.WHITE in self._attr_supported_color_modes and self._device.get("warm_dim", False):
            return ColorMode.WHITE
        elif ColorMode.COLOR_TEMP in self._attr_supported_color_modes and isinstance(current_color, WarmCoolColorValue):
            return ColorMode.COLOR_TEMP
        elif ColorMode.HS in self._attr_supported_color_modes and isinstance(current_color, FullColorValue):
            return ColorMode.HS

        return None
```

Reply:
It would be cleaner if we could determine the color mode from the device data itself without having to check `self._attr_supported_color_modes`